include/seqan3/alphabet/composition.hpp

```diff
@@ -175,41 +175,41 @@ struct alphabet_composition :
     //!\brief An index sequence up to the number of contained letters.
     static constexpr auto positions = std::make_index_sequence<sizeof...(alphabet_types)>{};
 
-    //!\brief Implementation of to_integral().
+    //!\brief Implementation of to_rank().
     template <std::size_t ...idx>
-    constexpr integral_type to_integral_impl(std::index_sequence<idx...> const &) const
+    constexpr rank_type to_rank_impl(std::index_sequence<idx...> const &) const
     {
-        using seqan3::to_integral;
+        using seqan3::to_rank;
         if constexpr (sizeof...(idx) > 0)
         {
-            return to_integral(get<0>(*this)) +
-                   ((to_integral(get<idx + 1>(*this)) * cummulative_alph_sizes[idx]) + ...);
+            return to_rank(get<0>(*this)) +
+                   ((to_rank(get<idx + 1>(*this)) * cummulative_alph_sizes[idx]) + ...);
```

Comment:
The thing is that the formula for computing this is
rank(0)
rank(1) * cum_alph_size(0)
rank(2) * cum_alph_size(1)
...
So we would need to make cum_alph_size size n + 1 which would make this function a little shorter, but would make the code more difficult to understand, because cum_alph_size[i] would not refer to the cummulative size of the i-th alphabet anymore. But it would also make the other function body simpler... I will think about it. thanks for the pointer!